crates/ide-diagnostics/src/handlers/missing_unsafe.rs

```diff
@@ -9,11 +15,99 @@ pub(crate) fn missing_unsafe(ctx: &DiagnosticsContext<'_>, d: &hir::MissingUnsaf
         "this operation is unsafe and requires an unsafe function or block",
         ctx.sema.diagnostics_display_range(d.expr.clone().map(|it| it.into())).range,
     )
+    .with_fixes(fixes(ctx, d))
+}
+
+fn fixes(ctx: &DiagnosticsContext<'_>, d: &hir::MissingUnsafe) -> Option<Vec<Assist>> {
+    let root = ctx.sema.db.parse_or_expand(d.expr.file_id)?;
```

Comment:
```suggestion
    if d.expr.file_id.is_macro() {
        return None;
    }
    let root = ctx.sema.db.parse_or_expand(d.expr.file_id)?;
```
The fixit will not work correctly for macro expansions, so for the time being let's disable this until we have a decent way of handling these cases in general.